kahuna/public/js/search/query.js

```diff
@@ -144,10 +147,28 @@ query.controller('SearchQueryCtrl', [
 
 
     $scope.$watchCollection(() => ctrl.filter, onValChange(filter => {
+        let newFilter = undefined;
+        storage.setJs("isNonFree", ctrl.filter.nonFree ? ctrl.filter.nonFree : false, true);
+
         filter.uploadedBy = filter.uploadedByMe ? ctrl.user.email : undefined;
+        storage.setJs("isUploadedByMe", ctrl.filter.uploadedByMe, true);
+
         ctrl.collectionSearch = ctrl.filter.query ? ctrl.filter.query.indexOf('~') === 0 : false;
 
-        $state.go('search.results', filter);
+
+        const defaultNonFreeFilter = storage.getJs("defaultNonFreeFilter", true);
+        if (defaultNonFreeFilter && defaultNonFreeFilter.isDefault === true){
+          let newNonFree = defaultNonFreeFilter.isNonFree ? "true" : undefined;
+          if (newNonFree !== filter.nonFree){
+                storage.setJs("isNonFree", newNonFree ? newNonFree : false, true);
+                storage.setJs("isUploadedByMe", false, true);
+                storage.setJs("defaultNonFreeFilter", {isDefault: false, isNonFree: false}, true);
+          }
+          newFilter = {};
+          newFilter = Object.assign(filter, {nonFree: newNonFree, uploadedByMe: false, uploadedBy: undefined});
```

Comment:
A few questions:

- What is the purpose of the reassignment on L167? It looks to me like `newFilter` is immediately reassigned to the result of `Object.assign` on the next line.
- To double check – IIUC Object.assign will mutate `filter` and return the target object ([MDN]((https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign))), which will mean that at as a result of this expression, `newFilter` and `filter` will be identical.  Is the mutation of `filter` intended? If so, I think `newFilter` is redundant – passing `filter` to the `$state.go` call should be equivalent, as the variables reference the same data.

Reply:
@jonathonherbert yup,` Object.assign` will mutate the object not return a new object
I fixed it in the last commit.